moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -811,13 +811,25 @@ const LinkModel* RobotState::getRigidlyConnectedParentLinkModel(const std::strin
 {
   const moveit::core::LinkModel* link{ nullptr };
 
-  size_t idx = 0;
-  if ((idx = frame.find('/')) != std::string::npos)
+  if (getRobotModel()->hasLinkModel(frame))
+  {
+    link = getLinkModel(frame);
+    if (transform)
+      transform->setIdentity();
+  }
+  else if (hasAttachedBody(frame))
+  {
+    auto* body{ getAttachedBody(frame) };
+    link = body->getAttachedLink();
+    if (transform)
+      *transform = body->getPose();
+  }
+  else if (size_t idx = frame.rfind('/'); idx != std::string::npos)
```

Comment:
This only searches for the last slash. If the subframe has slashes too, this will fail (I added a corresponding test).
We should use the same (proven) logic as in `getFrameInfo()`.